File: api/internal/model/house_offer.go
package model

// OfferFlags представляет предпочтения арендодателя (кого он хочет)
type OfferFlags struct {
	Price      *int    `json:"price"`
	RoomsCount *int    `json:"rooms_count"`
	District   *string `json:"district"`

	Smoking        *bool         `json:"smoking,omitempty"`
	Children       *ChildrenEnum `json:"children,omitempty"`
	Pets           *PetsEnum     `json:"pets,omitempty"`
	OccupantsCount *int          `json:"occupants_count,omitempty"`
	NoiseLvl       *NoiseLvlEnum `json:"noise_lvl,omitempty"`
	WorksFromHome  *bool         `json:"works_from_home,omitempty"`
	Alcohol        *AlcoholEnum  `json:"alcohol,omitempty"`
	AgeMin         *int          `json:"age_min,omitempty"`
	AgeMax         *int          `json:"age_max,omitempty"`
	Sex            *SexEnum      `json:"sex,omitempty"`
}

type HouseOffer struct {
	Id             int64    `json:"id"`
	IsActive       bool     `json:"is_active"`
	OwnerId        int64    `json:"owner_id"`
	Description    string   `json:"description"`
	City           string   `json:"city"`
	MediaFiles     []string `json:"media_files"`
	FlagProcessing bool     `json:"flag_processing"`

	// Флаги объявления
	OfferFlags `json:"flags"`
}

// HouseOfferCreate представляет данные, необходимые для создания объявления.
type HouseOfferCreate struct {
	OwnerId     int64    `json:"owner_id" binding:"required"`
	Description string   `json:"description" binding:"required"`
	City        string   `json:"city" binding:"required"`
	MediaFiles  []string `json:"media_files" binding:"required"`
}

// HouseOfferPreview представляет поверхностные данные, которые видит владелец своих объявлений.
type HouseOfferPreview struct {
	Id         int64  `json:"id"`
	IsActive   bool   `json:"is_active"`
	Title      string `json:"title"`
	LikesCount int    `json:"likes_count"`
}

type AddLikeRequest struct {
	OfferId   int64 `json:"offer_id"`
	UserId    int64 `json:"user_id"`
	Relevance int   `json:"relevance"`
}

// HouseOfferLike представляет данные о лайке объявления.
type HouseOfferLike struct {
	Id        int64 `json:"id"`
	OfferId   int64 `json:"offer_id"`
	UserId    int64 `json:"user_id"`
	Relevance int   `json:"relevance"`
}

type RandRelevantOfferRequest struct {
	UserID    int64     `json:"user_id" binding:"required"`
	City      string    `json:"city" binding:"required"`
	UserFlags UserFlags `json:"user_flags"`
}

type RelevantOffer struct {
	RelevanceSum     int `json:"relevance_sum"`
	RelevancePercent int `json:"relevance_percent"`
	HouseOffer       `json:"offer"`
}

File: bot/app/model/house_offer.py
from dataclasses import dataclass, field
from .enums import *

@dataclass
class OfferFlags:
    """Флаги объявления"""
    price: int | None = 0
    rooms_count: int | None = 0
    district: str | None = ""

    smoking: bool | None = None
    children: ChildrenEnum | None = None
    pets: PetsEnum | None = None
    occupants_count: int | None = None
    noise_lvl: NoiseLvlEnum | None = None
    works_from_home: bool | None = None
    alcohol: AlcoholEnum | None = None
    age_min: int | None = None
    age_max: int | None = None
    sex: SexEnum | None = None


@dataclass
class HouseOffer:
    """Данные объявления о сдаче/продаже жилья."""
    id: int = 0
    owner_id: int = 0
    is_active: bool = False
    description: str = ""
    city: str = ""
    media_files: list[str] = field(default_factory=list)
    flag_processing: bool = False
    flags: OfferFlags = field(default_factory=OfferFlags)


@dataclass
class HouseOfferCreate:
    """Данные, необходимые для создания объявления."""
    owner_id: int = 0
    description: str = ""
    city: str = ""
    media_files: list[str] = field(default_factory=list)


@dataclass
class HouseOfferPreview:
    """Поверхностные данные, которые видит владелец своих объявлений."""
    id: int = 0
    is_active: bool = False
    title: str = ""
    likes_count: int = 0


@dataclass
class AddLikeRequest:
    offer_id: int = 0
    user_id: int = 0
    relevance: int = 0

@dataclass
class HouseOfferLike:
    id: int = 0
    offer_id: int = 0
    user_id: int = 0
    relevance: int = 0


@dataclass
class RelevantOffer:
    relevance_sum: int = 0
    relevance_percent: int = 0
    offer: HouseOffer = field(default_factory=HouseOffer)
